perf_test/Benchmark_Context.hpp

```diff
@@ -95,6 +95,24 @@ inline void add_benchmark_context(bool verbose = false) {
   add_version_info();
 }
 
+inline void register_benchmark(const char* name,
+                               void (*func)(benchmark::State&),
```

Comment:
I think one problem with this is that we are constraining to functions that only accept state. 
What if we needed to pass functions accepting multiple extra arguments? 
Can we make this a variadic thing like:
```cpp
template<class FuncType, class ... ArgsToCallOp>
void register_benchmark(const char* name, FuncType func, 
                                            std::vector<std::string> arg_names, 
                                            std::vector<int64_t> args, 
                                            int repeat,
                                           ArgsToCallOp && ... args)
{
  // if ()
  // ...
  benchmark::RegisterBenchmark(name, func, std::forward<ArgsToCallOp>(args)...);
}
```

Reply:
Done.